File: server/autogen_service/list_models.py
import os
from google import genai
from dotenv import load_dotenv

parent_env_path = os.path.join(os.path.dirname(__file__), '..', '.env')
load_dotenv(parent_env_path)

api_key = os.getenv('GEMINI_API_KEY')
if not api_key:
    print("No API key found in .env!")
    exit(1)

client = genai.Client(api_key=api_key)

print("Models supporting generateContent:")
try:
    for m in client.models.list():
        print(f"- {m.name}")
except Exception as e:
    print(f"Error listing models: {e}")

File: server/list_models.js
import dotenv from 'dotenv';
import axios from 'axios';

dotenv.config();

async function listModels() {
    const API_KEY = process.env.GEMINI_API_KEY;
    try {
        const response = await axios.get(`https://generativelanguage.googleapis.com/v1beta/models?key=${API_KEY}`);
        console.log("Available Models:");
        response.data.models.forEach(m => {
            console.log(`- ${m.name} (${m.supportedGenerationMethods.join(', ')})`);
        });
    } catch (error) {
        console.error("Error listing models:", error.response?.data || error.message);
    }
}

listModels();
